### PR TITLE
adds forward proxy support

### PR DESCRIPTION
**Pull request by amprokop**
Connects https://github.com/department-of-veterans-affairs/ruby-bgs/issues/43

Testing:
Installed this gem on Caseflow Monitor preprod, requests worked fine.

### CODE REVIEW

**Comment by lisac-usds** [comment]
Hello @aroltsch , would you please review this modification? thanks.

File: lib/bgs/base.rb [hunk 1]
```diff
@@ -110,6 +132,8 @@ def client
 
     # Proxy to call a method on our web service.
     def request(method, message = nil)
+      # can be removed when savon > 2.11.2 is released
+      client.wsdl.request.headers = { "Host" => domain } if @forward_proxy_url
```
Comment (aroltsch) [inline lib/bgs/base.rb]: Is it not working when including `headers` on line 125? @lisac-usds

File: lib/bgs/base.rb [hunk 2]
```diff
@@ -98,9 +114,15 @@ def header
     # logging can be enabled by passing `log: true` to the constructor
     # of any of the services.
     def client
+      # Tack on the destination header if we're sending all requests
+      # to a forward proxy.
+      headers = {}
+      headers["Host"] = domain if @forward_proxy_url
```
Comment (aroltsch) [inline lib/bgs/base.rb]: Cool

File: lib/bgs/base.rb (hunk 1 above)
Comment (lisac-usds) [reply lib/bgs/base.rb]: @aroltsch that's correct. The logic on line 125 should suffice with the next release of savon - it introduces the header support to `WSDLRequest` (the change [here](https://github.com/savonrb/savon/blob/73eb888b7f896d5bd0551971d26436b4eb90afd2/lib/savon/request.rb#L61). 

In the meantime, we still need to set a host header, in order for the forward proxy to field the request. The approach we took is line 136.

**Comment by kierachell** [comment]
- Can we adjust the client to set `forward_proxy_url` from the env? they way it is done here:
https://github.com/department-of-veterans-affairs/connect_vbms/blob/master/lib/vbms/client.rb#L43
Tested in UAT:
```
[2017-11-18T19:42:08.164Z] "GET /OrgWebServiceBean/OrgWebService?WSDL HTTP/1.1" 200 - 0 4810 207 207 "-" "HTTPClient/1.0 (2.8.3, ruby 2.3.0 (2015-12-25))" "ac787103-ee3b-44b2-9271-2a71313cadca" "beplinktest.vba.va.gov" "152.131.36.131:443"
[2017-11-18T19:42:08.308Z] "GET /AddressWebServiceBean/AddressWebService?WSDL HTTP/1.1" 200 - 0 8792 228 151 "-" "HTTPClient/1.0 (2.8.3, ruby 2.3.0 (2015-12-25))" "ea9c3944-14d2-400c-8059-d437e4b2556b" "beplinktest.vba.va.gov" "152.131.36.131:443"
[2017-11-18T19:42:08.366Z] "GET /BenefitClaimServiceBean/BenefitClaimWebService?WSDL HTTP/1.1" 200 - 0 10774 277 130 "-" "HTTPClient/1.0 (2.8.3, ruby 2.3.0 (2015-12-25))" "d5594a18-ba34-4ad9-a707-aaaa6f241cf5" "beplinktest.vba.va.gov" "152.131.36.131:443"
[2017-11-18T19:42:08.740Z] "GET /VetRecordServiceBean/VetRecordService?WSDL HTTP/1.1" 200 - 0 11243 471 324 "-" "HTTPClient/1.0 (2.8.3, ruby 2.3.0 (2015-12-25))" "ddc66290-4e61-443d-906b-155416034b5b" "beplinktest.vba.va.gov" "152.131.36.131:443"
```

File: lib/bgs/base.rb (hunk 1 above)
Comment (amprokop) [reply lib/bgs/base.rb]: 👍

**Comment by amprokop** [comment]
@lisac-usds feel free to merge this while i'm away

**Pull request merged**